### DISCUSSION TITLE
Static files are not loaded generating pdf from html when application run in ASGI layer

### DISCUSSION THREAD

**Discussion by tanvir0ahmmed** [django/channels#1847] [Help!] [answered]
**I use Weasyprint to generate pdf from HTML. When run the application in ASGI static files are not loaded in pdf, but if I run the application in WSGI then the static file loads correctly. How can I solve this issue or can I use StaticFilesWrapper this class to solve this issue, how?**
![image](https://user-images.githubusercontent.com/60448694/169695552-cd833df2-aa4d-4913-b7b0-33a9b1a3446c.png)
![image](https://user-images.githubusercontent.com/60448694/169695722-fa4961e2-7c4a-42c6-bd02-4de3f7d31d7a.png)
![image](https://user-images.githubusercontent.com/60448694/169695754-9e3e0a24-dbac-4bca-9dcd-20670cb546ad.png)

**Comment by carltongibson** [comment, marked as answer]
Hi @tanvir0ahmmed. Please don't double post. 

This could be related to #1722, but there's not really enough information to see. (Addressing #1722 properly is on the current TODO list.)

Generally one doesn't have Django serve static files except in development: see [How to Deploy Static Files | Django Documentation | Django](https://docs.djangoproject.com/en/4.0/howto/static-files/deployment/)
